File: programmers/js/new/level2/큰_수_만들기.js
function solution(number, k) {
    var answer = "";
    const stack = [];
    number.split("").forEach((num) => {
        const n = parseInt(num);
        if (stack.length === 0) {
            stack.push(n);
        } else {
            let top = stack[stack.length - 1];
            while (top < n && k > 0) {
                stack.pop();
                top = stack[stack.length - 1];
                k--;
            }
            if (stack.length < number.length - k) {
                stack.push(n);
            }
        }
    });
    return stack.join("");
}

File: programmers/python/level2/큰_수_만들기.py
from collections import deque
def solution(number, k):
    st = deque()
    for num in number:
        n = int(num)
        if len(st) == 0:
            st.append(n)
        else:
            while len(st) > 0 and st[-1] < n and k > 0:
                st.pop()
                k -= 1
            st.append(n)
    while k > 0:
        st.pop()
        k -= 1
    return ''.join(map(str,st))

print(solution('1924', 2))